### ISSUE TITLE
Flutter Doctor reports Android SDK toolchain  wrong location

### ISSUE THREAD

**Issue by Buddyboy-git** [flutter/flutter#139865]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Fresh install of flutter and Android Studio.
Run Flutter Doctor.
Flutter expects the Android SDK to be int the LIB folder under Android Studio.
However Android Studio does NOT allow the SDK to exist inside of Android Studio directory.

### Actual results

Flutter expects the Android SDK to be int the LIB folder under Android Studio.
However Android Studio does NOT allow the SDK to exist inside of Android Studio directory.

### Logs

<details>
<summary>Logs</summary>

```console
no logs

<!-- Paste your logs here -->

```

</details>

### Flutter Doctor output

<details>
<summary>Doctor output</summary>

```console

<!-- Paste your output here -->

```
D:\>flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.16.3, on Microsoft Windows [Version 10.0.22621.2715], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[X] Android toolchain - develop for Android devices
    X ANDROID_HOME = D:\Program Files\Android\Android Studio\plugins\android\lib
      but Android SDK not found at this location.
[√] Chrome - develop for the web
[√] Visual Studio - develop Windows apps (Visual Studio Build Tools 2019 16.11.19)
[√] Android Studio (version 2023.1)
[√] VS Code (version 1.84.2)
[√] Connected device (3 available)
[√] Network resources

! Doctor found issues in 1 category.
</details>

**Comment by huycozy** [comment]
It seems you are setting the wrong ANDROID_HOME path. Can you please share Android SDK path? And please share the output of `flutter doctor -v` as well. Thanks!

**Comment by Buddyboy-git** [comment]
Sorry. As a noob, I did not set a HOMEPATH. 
And I honestly don't think it should matter. Android Studio simply rejects Adroid SDK to be under its installed directory.  If you move the SDK to where flutter expects it to be , The Studio will simply not let you go any further. And Flutter defaults to a LIB directory under  the Android  Studio install.
Step 1: Install Android Studio as is, it now auto installs the SDK.
Step 2: Install Flutter as is.
Step 3. Run Docter.

I fixed it by running Flutter configuration to point to proper SDK location. 

But no matter what the Install is... Flutter should not be looking for the SDK inside of the Studio directory.

**Comment by Buddyboy-git** [comment]
Yes. Thankyou. I and 100s of others have eventually solved this issue...
I was just trying to make the flutter team aware that the flutter default
is wrong. The SDK  is NOT allowed to be under the studio directory.
Therefore there are no circumstances where flutter should default to look
there....

**Comment by huycozy** [comment]
Weirdly, it's looking at that path, it should be something like this by default: `C:\Users\<username>\AppData\Local\Android\sdk\`:

https://github.com/flutter/flutter/blob/3b370c8539a4566d1ec1145000ebcd134c34552b/packages/flutter_tools/lib/src/android/android_sdk.dart#L109-L119

What are your ANDROID_HOME and Android SDK paths? 

Did you install Android SDK inside `D:\Program Files\Android\Android Studio\` directory?

**Comment by Buddyboy-git** [comment]
I was simply stating....
That a  fresh  Android Studio installation will put the SDK ANYWHERE EXCEPT under its own directory. ...   It has its own set of rules for some reason that the SDK Can't be inside if the Studio installation directory....

And then when you do a fresh install of flutter, it s default configuration expects the Android SDK to be exactly where Android Studio does not allow it to be...  which is UNDER or INSIDE the Android Studio directory.  

So In conclusion...  No matter what configurations or paths can be changed prior to or after both installations.    FLUTTER should never, ever expect the Android SDK to be inside of the Android Studio directory. But that's exactly where it looks by default....

And that is why so many people have issues after installing flutter.

**Comment by huycozy** [comment]
>And then when you do a fresh install of flutter, it s default configuration expects the Android SDK to be exactly where Android Studio does not allow it to be... which is UNDER or INSIDE the Android Studio directory.

>And that is why so many people have issues after installing flutter.

I tried searching around but couldn't find the same case in which the ANDROID_HOME  like yours above. Could you please share some reports you found? 

Also, did you set that ANDROID_HOME in your environment system variables? Please help to answer questions above so that the issue can be more clear to move forward. Thanks!
